### ISSUE TITLE
Possible solution for inner and outer local objects

### ISSUE THREAD

**Issue by bigodel** [noctuid/lispyville#302]
on line 2036 there is the following TODO:

https://github.com/noctuid/lispyville/blob/32d6b1f4b3f5bc792a0610c7e91123a70b582a91/lispyville.el#L2036-L2050

looking that the [documentation for `evil-surround`](https://github.com/emacs-evil/evil-surround#add-surround-pairs-for-buffer-local-text-objects) i saw that they have a way to define local inner and outer text objects. i haven't been deep into their code, but it seems like `lispyville` could take a look into it to try and replicate. i opened up the issue in order to keep track of it, and i might tackle it if i manage to get some free time.

**Comment by noctuid** [comment]
Yes you can do buffer-local but not mode-local text objects.  There is an issue somewhere where I evaluated different methods of working around this.  The current suggestion in the lispyville readme is to bind in operator/visual state instead of those keymaps.  Assuming that works, I think that would be the better solution.
